### ISSUE TITLE
[java] AddEmptyString - false negative with empty var

### ISSUE THREAD

**Issue by Belle-PL** [pmd/pmd#3625]
**Affects PMD Version:** 
6.39.0

**Rule:**
AddEmptyString

Please provide the rule name and a link to the rule documentation:
<https://pmd.github.io/latest/pmd_rules_java_performance.html#addemptystring>

**Description:**
PMD should have reported a warning at line 3, because String variable `test` adds an empty string, but it doesn't. This can lead to performance issue. So, I think it is a false negative. Thanks for your consideration.
**Code Sample demonstrating the issue:**

```java
public void func() {
    final String var = "";
    String test = var + "123";  // should report a warning here
}
```

**Expected outcome:**

PMD should report a violation at line 3, but doesn't. This is a false-negative.

**Running PMD through:** *[Maven]*

**Comment by LiGaOg** [comment]
Please assign this issue to me
